sdk/containerregistry/Azure.Containers.ContainerRegistry/src/Authentication/ContainerRegistryChallengeAuthenticationPolicy.cs

```diff
@@ -102,11 +112,26 @@ private async Task<string> ExchangeAcrRefreshTokenForAcrAccessTokenAsync(string
             Response<AcrAccessToken> acrAccessToken = null;
             if (async)
             {
-                acrAccessToken = await _authenticationClient.ExchangeAcrRefreshTokenForAcrAccessTokenAsync(service, scope, acrRefreshToken, cancellationToken).ConfigureAwait(false);
+                acrAccessToken = await _authenticationClient.ExchangeAcrRefreshTokenForAcrAccessTokenAsync(service, scope, acrRefreshToken, cancellationToken: cancellationToken).ConfigureAwait(false);
+            }
+            else
+            {
+                acrAccessToken = _authenticationClient.ExchangeAcrRefreshTokenForAcrAccessToken(service, scope, acrRefreshToken, cancellationToken: cancellationToken);
+            }
+
+            return acrAccessToken.Value.AccessToken;
+        }
+
+        private async Task<string> GetAnonymousAcrAccessTokenAsync(string service, string scope, bool async, CancellationToken cancellationToken)
+        {
+            Response<AcrAccessToken> acrAccessToken = null;
+            if (async)
+            {
+                acrAccessToken = await _authenticationClient.ExchangeAcrRefreshTokenForAcrAccessTokenAsync(service, scope, acrRefreshToken: string.Empty, TokenGrantType.Password, cancellationToken ).ConfigureAwait(false);
```

Comment:
should we move acrRefreshToken to the first arg so that we don't need to name it?

Reply:
I just had it there because I was passing null previously and wanted to disambiguate what the parameter was.  Happy to take it out.